frame/dvm/src/lib.rs

```diff
@@ -351,13 +355,20 @@ impl<T: Config> Module<T> {
 		}
 	}
 
-	fn do_transact(transaction: ethereum::Transaction) -> DispatchResultWithPostInfo {
+	pub fn do_transact(
+		transaction: ethereum::Transaction,
+		sender: Option<H160>,
+	) -> DispatchResultWithPostInfo {
 		ensure!(
 			dp_consensus::find_pre_log(&<frame_system::Pallet<T>>::digest()).is_err(),
 			Error::<T>::PreLogExists,
 		);
-		let source =
-			Self::recover_signer(&transaction).ok_or_else(|| Error::<T>::InvalidSignature)?;
+
+		// For transaction from ethereum issuing pallet, the sender set None to skip the signature verification.
+		let source = sender.map_or(
+			Self::recover_signer(&transaction).ok_or_else(|| Error::<T>::InvalidSignature)?,
+			|source| source,
+		);
```

Comment:
This implementation looks dirty.

I think we can safely remove this part of codes since these checks have already be done in `ValidateUnsigned`.

Other internal code from other pallet is relying on audit for safety.

Reply:
remove param sender and use address 0x0 as a runtime common caller address.